protoc-gen-flowtypes/genflowtypes/genflowtypes.go

```diff
@@ -36,71 +37,78 @@ var knownTypeMap = map[string]string{
 	".google.protobuf.Timestamp": "string",
 }
 
-func newSimpleType(typeString string, required bool) *primitiveType {
-	return &primitiveType{typeString, required}
+func newSimpleType(typeString string, opts opts.Options) *primitiveType {
+	return &primitiveType{typeString, opts}
 }
 
 type primitiveType struct {
 	typeString string
-	required   bool
+	opts       opts.Options
 }
 
 func (s primitiveType) FlowType() string { return s.typeString }
-func (s primitiveType) IsRequired() bool { return s.required }
+func (s primitiveType) IsRequired() bool { return s.opts.GetRequired() }
+func (s primitiveType) IsNullable() bool { return s.opts.GetNullable() }
 
 type messageType struct {
 	*primitiveType
 }
 
-func newMessageFlowType(typeName string, required bool) *messageType {
-	s := newSimpleType(typeName, required)
+func newMessageFlowType(typeName string, opts opts.Options) *messageType {
+	s := newSimpleType(typeName, opts)
 	return &messageType{s}
 }
 
 type repeatedType struct {
 	FlowTyper
-	required bool
+	opts opts.Options
 }
 
-func newRepeatedFlowType(underlying FlowTyper, required bool) *repeatedType {
-	return &repeatedType{FlowTyper: underlying, required: required}
+func newRepeatedFlowType(underlying FlowTyper, opts opts.Options) *repeatedType {
+	return &repeatedType{FlowTyper: underlying, opts: opts}
 }
 
 func (r repeatedType) FlowType() string { return fmt.Sprintf("Array<%s>", r.FlowTyper.FlowType()) }
 
 type namedType struct {
 	FlowTyper
-	name     string
-	required bool
+	name string
+	opts opts.Options
 }
 
 func (t *namedType) Name() string {
 	return t.name
 }
 
 type objectFlowType struct {
-	Fields   []NamedFlowTyper
-	Options  Options
-	required bool
+	Fields  []NamedFlowTyper
+	Options Options
+
+	opts opts.Options
```

Comment:
This struct has `Options` and `opts`, is there some way this confusing naming could be avoided? `FlowOptions`? I'm not familiar enough with the code but this stuck out to me as confusing.

Reply:
Good point. I’ll consider a clarity pass.